### PR TITLE
Remove set_parameters and data setter from qscript

### PR DESCRIPTION
**Pull request by lillian542**
Completes deprecation cycle for `QuantumScript.set_parameters` and `QuantumScript.data`

### CODE REVIEW

File: doc/development/deprecations.rst [hunk 1]
```diff
@@ -113,8 +110,12 @@ Pending deprecations
 Completed deprecation cycles
 ----------------------------
 
+* The ``QuantumScript.set_parameters`` method and the ``QuantumScript.data`` setter have
+  been removed. Please use ``QuantumScript.bind_new_parameters`` instead.
+
 * The CV observables ``qml.X`` and ``qml.P`` have been removed. Use ``qml.QuadX`` and ``qml.QuadP`` instead.
 
+
```
Comment (lillian542) [inline doc/development/deprecations.rst:118]:
```suggestion
```

**Comment by lillian542** [comment]
[sc-43899]

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -21,6 +21,10 @@
   Instead of ``tape.unwrap()``, use :func:`~.transforms.convert_to_numpy_parameters`.
   [#4535](https://github.com/PennyLaneAI/pennylane/pull/4535)
 
+* The ``QuantumScript.set_parameters`` method and the ``QuantumScript.data`` setter have
+  been removed. Please use ``QuantumScript.bind_new_parameters`` instead.
+  [#4548](https://github.com/PennyLaneAI/pennylane/pull/4548)
```
Comment (timmysilv) [inline doc/releases/changelog-dev.md:26]:
```suggestion
  [(#4548)](https://github.com/PennyLaneAI/pennylane/pull/4548)
```

**Review by timmysilv — APPROVED** [review]

**Review by mudit2812 — APPROVED** [review]

File: tests/tape/test_tape.py [hunk 3]
```diff
@@ -851,244 +851,6 @@ def test_measurement_parameter(self):
...
-        tape, circuit_params, hermitian = make_tape_with_hermitian
-        num_all_params = len(circuit_params) + 1  # + 1 for hermitian
-        assert tape.num_params == num_all_params
-        assert tape.trainable_params == list(range(num_all_params))
-        assert (
-            tape.get_parameters(operations_only=operations_only) == circuit_params
-            if operations_only
-            else circuit_params + [hermitian]
-        )
-
```
[the remark this answers is not in the document]
Comment (lillian542) [reply tests/tape/test_tape.py:906]: Even the ones not using `set_parameters` are duplicated in the `TestParameters` class, so while valid, the `get_parameters` tests here shouldn't be needed

File: pennylane/tape/qscript.py [hunk 4]
```diff
@@ -19,7 +19,6 @@
 
 import contextlib
 import copy
```
Comment (lillian542) [inline pennylane/tape/qscript.py:21]: import warnings

File: pennylane/tape/qscript.py [hunk 5]
```diff
@@ -19,7 +19,6 @@
...
-import warnings
 from collections import Counter
```
Comment (lillian542) [inline pennylane/tape/qscript.py:22]:
```suggestion
import warnings
from collections import Counter
```

File: tests/tape/test_tape.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (albi3ro) [reply tests/tape/test_tape.py:906]: sounds good.

**Pull request merged**